### PR TITLE
DS-3806 Subscript span attr for 1d and 2d tables

### PR DESCRIPTION
**Pull request by jrwishart**
- DS-3806 Add basic testing function for attributes
- DS-3806 Add temporary test case
- DS-3806 Add template for attribute update [revdep skip]
- DS-3806 Evalaute arguments in correct frame
- DS-3806 Use evaluated args in span [revdep skip]
- DS-3806 Extend to 2d and 2d with multiple stats
- DS-3806 Simplify test code calls
- DS-3806 Retain spans with character matching [revdep skip]

### CODE REVIEW

File: R/subscript-tables.R [hunk 1]
```diff
@@ -147,7 +155,29 @@ updateTableAttributes <- function(y, x, called.args) {
...
+        idx <- which(span.attr[[2]] %in% idx)
+    out <- span.attr[idx, , drop = FALSE]
+    if (all(is.na(out[[1L]]))) invisible() else out
+}
+
+updateSpanIfNecessary <- function(y, x.attributes, called.args, evaluated.args) {
+    span.attribute <- x.attributes[["span"]]
+    if (is.null(span.attribute)) return(y)
+    dim.length <- length(x.attributes[["dim"]])
+    if (dim.length > 1L && length(evaluated.args) == 1L) return(y)
```
Comment (jrwishart) [inline R/subscript-tables.R:175]: Will add comment about single integer vector will drop spans if array has more than one dim.

File: R/subscript-tables.R [hunk 2]
```diff
@@ -147,7 +155,29 @@ updateTableAttributes <- function(y, x, called.args) {
...
     names.needing.update <- !isBasicAttribute(attr.names)
     names(attributes(y))[names.needing.update] <- paste0("original.", attr.names[names.needing.update])
+    y <- updateSpanIfNecessary(y, x.attributes, called.args, evaluated.args)
     attr(y, "name") <- paste0(x.attributes[["name"]], "[",
                               paste(as.character(called.args), collapse = ","), "]")
     y
 }
+
+subscriptSpanDF <- function(span.attr, idx) {
+    if (is.character(idx))
```
Comment (jrwishart) [inline R/subscript-tables.R:165]: Add empty span check (only one column in span df) and return empty early

File: R/subscript-tables.R (hunk 1 above)
Comment (jrwishart) [reply R/subscript-tables.R:175]: Add support for integer matrix (with d columns (1 or 2d)) and boolean matrix (only if bool mat is same size)

File: R/subscript-tables.R (hunk 2 above)
Comment (jrwishart) [reply R/subscript-tables.R:166]: Resolved in https://github.com/Displayr/verbs/pull/43/commits/aaba8b4a9f1a0a07f9e7c48b88b9b5771903fc3a

File: R/subscript-tables.R (hunk 1 above)
Comment (jrwishart) [reply R/subscript-tables.R:181]: Dropping support for integer matrices and boolean matrices. They don't retain their dimension in base R with that approach. 
E.g. the subscriptted array below will be a vector and not a matrix with col names and row names
```
x <- array(1:12, dim = 3:4, dimnames = list(LETTERS[1:3], letters[1:4]))
ind <- array(FALSE, dim = 3:4)
ind[1:2, ] <- TRUE
x[ind]
x[1:2, ]
```

**Review by mwmclean — APPROVED** [review]

**Pull request merged**